### ISSUE TITLE
Feature request: giving more control on the way variables are named

### ISSUE THREAD

**Issue by fblanqui** [rlepigre/ocaml-bindlib#11]
Currently, a variable is represented as a record as follows:
https://github.com/rlepigre/ocaml-bindlib/blob/344937fb4a1b6f42d01469d95c5550db51892f02/bindlib.ml#L132-L137
When a new variable is created, its prefix and suffix are computed with the function split_name as follows:
https://github.com/rlepigre/ocaml-bindlib/blob/344937fb4a1b6f42d01469d95c5550db51892f02/bindlib.ml#L147-L152
The prefix remains unchanged. On the other hand, the suffix may be incremented.
Finally, the name of variable is built by concatenating its prefix and its suffix:
https://github.com/rlepigre/ocaml-bindlib/blob/344937fb4a1b6f42d01469d95c5550db51892f02/bindlib.ml#L142-L145
This may create problems if the set of valid names we work with is not closed by concatenation.
This could be fixed by providing a new function, say name_of_with, that would take a user-defined function merge_name as additional argument:
```
let name_of_with : (string -> int -> string) -> 'a var -> string = fun merge_name x -> merge_name x.var_prefix x.var_suffix
```

**Comment by fblanqui** [comment]
The fact that leading zeros in integer suffixes are ignored may also be a source of problems as an expression generated by Bindlib may not have the same meaning as the one that has been parsed. Take for instance (lambda x01, x01 x1), where x1 is a constant. Then, Bindlib would rename the bound variable x01 into x1.
So, it would be convenient that leading zeros are included in the prefix.

**Comment by craff** [comment]
In the example you suggest above, bindlib would rename the term 
to lambda x2, x2 x1 if well used. So this is not really a problem to ignore leading zeros.

**Comment by craff** [comment]
Remark: you have two ways to use bindlib: do renaming at term construction (increase the cost of computation, and adds extra term traversal) or at printing time (probably the recommended way), using context. We should check the documentation about this. And it should also be very clear that bindlib uses name only for printing and by the way, in the old time, there was a version of bindlib with no name. 

Actually, the best solution is probably no name at all in terms and use of context mandatory for printing. Then the renaming policy could by much more customizable and would have no impact on computation (only impact printing).

**Comment by craff** [comment]
PR #12 proposes a solution

**Comment by rlepigre** [comment]
Fixed by c1fa429388d1faeea57e3ee3cd29d92810b62719, which will be part of the next release of Bindlib (version 6.0.0).
